src/main/java/com/glencoesoftware/omero/ms/image/region/PixelsService.java

```diff
@@ -100,6 +99,10 @@ private Path asPath(String ngffDir) throws IOException {
 
         try {
             URI uri = new URI(ngffDir);
+            Map<String, String> params = Splitter.on('&')
+                    .trimResults()
+                    .withKeyValueSeparator('=')
+                    .split(uri.getQuery());
```

Comment:
If there are no query strings, getQuery returns null, which results in a  NullPointerException. Maybe just move this inside the s3 conditional?

Reply:
Good point. It definitely need to go into the S3 conditional and handle `null` adequately. I made a bad assumption that it would return `""`.